### ISSUE TITLE
Change swagger baseUrl in a microservices environment

### ISSUE THREAD

**Issue by peppa2016** [springfox/springfox#2145]
We are developing a few spring boot application conforming to microservices architecture. The gateway being used in zuul netflix. We were able to bring up the swagger page for the gateway. The issue is that we would like to change the basePath of one of the microservices application. Currently the basePath is getting picked up from the zuul routes. Something like
zuul:
routes:
foo-services:
serviceId: foo-services
path: /foo/**
So the basePath is shown as /foo. But can I change it to only "/"
springBootVersion = '1.5.3.RELEASE'
springfoxVersion = '2.6.1'

Thanks.

**Comment by dilipkrish** [comment]
You need to set it up so that null forwards the X-Forwarded... headers. See here [for an explanation](https://cloud.spring.io/spring-cloud-netflix/multi/multi__router_and_filter_zuul.html#netflix-zuul-reverse-proxy).

Also please move to using springfox 2.7.0

**Comment by peppa2016** [comment]
Thanks for the quick reply. Actually we already have a route at null. So,
zuul:
 routes:
  foo-services:
    serviceId: foo-services
    path: /foo/**
 bar-services:
    serviceId: bar-services
    path: /**
.....
The basePath for bar-services is / , can the basePath for foo-services be / as well?

**Comment by dilipkrish** [comment]
You need to get set the X-Forward Headers.

**Comment by peppa2016** [comment]
Sorry I am new to all these, ll u be able to explain a bit more.

**Comment by dilipkrish** [comment]
@peppa2016 use the link I posted ☝️ It has a good explanation. Plus its the source of Zuul

**Comment by peppa2016** [comment]
As provided in the link : The prefix path is stripped by default, and the request to the backend picks up a header "X-Forwarded-Prefix". So the value taken in by X-Forwarded-Prefix is the route.getPrefix() value, as explained [here](https://github.com/spring-cloud/spring-cloud-netflix/issues/1445). So I tried setting route.setPrefix(""), so that the base Url is /, but it ends up being /foo again. Any pointers? Thanks again.

**Comment by peppa2016** [comment]
I fixed this quite sometime back but couldnt reply here. As pointed by @dilipkrish I had to set the X-Forwarded-Prefix. I did that in a custom zuul pre-filter. Since the link above doesnt talk much about setting the headers, this is what I did,
`public Object run() {
        final RequestContext requestContext = RequestContext.getCurrentContext();
requestContext.addZuulRequestHeader("X-Forwarded-Prefix", "/"); ...
}
`
Thanks for the pointers @dilipkrish
